src/sagas/repository.js

```diff
@@ -345,6 +345,32 @@ export function* getDatasetTablePreview({ payload }) {
   }
 }
 
+/**
+ * billing profile
+ */
+export function* getBillingProfileById(profileId) {
+  try {
+    const response = yield call(authGet, `/api/resources/v1/profiles/${profileId}`);
+    yield put({
+      type: ActionTypes.GET_BILLING_PROFILE_BY_ID_SUCCESS,
+      profile: { data: response },
+    });
+  } catch (err) {
+    yield put({
+      type: ActionTypes.EXCEPTION,
+      profile: null,
+    });
+  }
+}
+
+export function* watchRequests() {
```

Comment:
i'm not super familiar with `take` and `actionChannel`, but it might be useful to give this a more billing profile specific name unless we plan on adding other polling actions here as well